streams/src/test/java/org/apache/kafka/streams/integration/AbstractJoinIntegrationTest.java

```diff
@@ -229,92 +178,50 @@ void runTestWithDriver(final List<List<TestRecord<Long, String>>> expectedResult
                     expectedFinalResult = updatedExpected.get(expected.size() - 1);
                 }
             }
-        }
-    }
-
-
-    /*
-     * Runs the actual test. Checks the result after each input record to ensure fixed processing order.
-     * If an input tuple does not trigger any result, "expectedResult" should contain a "null" entry
-     */
-    void runTest(final List<List<KeyValueTimestamp<Long, String>>> expectedResult, final String storeName) throws Exception {
-        assert expectedResult.size() == input.size();
-
-        IntegrationTestUtils.purgeLocalStreamsState(STREAMS_CONFIG);
-        streams = new KafkaStreams(builder.build(), STREAMS_CONFIG);
-
-        KeyValueTimestamp<Long, String> expectedFinalResult = null;
-
-        try {
-            startKafkaStreamsAndWaitForRunningState(streams, TIMEOUT);
-
-            final long firstTimestamp = System.currentTimeMillis();
-            long ts = firstTimestamp;
-
-            final Iterator<List<KeyValueTimestamp<Long, String>>> resultIterator = expectedResult.iterator();
-            for (final Input<String> singleInput : input) {
-                producer.send(new ProducerRecord<>(singleInput.topic, null, ++ts, singleInput.record.key, singleInput.record.value)).get();
-
-                final List<KeyValueTimestamp<Long, String>> expected = resultIterator.next();
-
-                if (expected != null) {
-                    final List<KeyValueTimestamp<Long, String>> updatedExpected = new LinkedList<>();
-                    for (final KeyValueTimestamp<Long, String> record : expected) {
-                        updatedExpected.add(new KeyValueTimestamp<>(record.key(), record.value(), firstTimestamp + record.timestamp()));
-                    }
-
-                    checkResult(OUTPUT_TOPIC, updatedExpected);
-                    expectedFinalResult = updatedExpected.get(expected.size() - 1);
-                }
-            }
 
             if (storeName != null) {
-                checkQueryableStore(storeName, expectedFinalResult);
+                checkQueryableStore(storeName, expectedFinalResult, driver);
             }
-        } finally {
-            streams.close();
         }
     }
 
-    /*
-     * Runs the actual test. Checks the final result only after expected number of records have been consumed.
-     */
-    void runTest(final KeyValueTimestamp<Long, String> expectedFinalResult, final String storeName) throws Exception {
-        IntegrationTestUtils.purgeLocalStreamsState(STREAMS_CONFIG);
-        streams = new KafkaStreams(builder.build(), STREAMS_CONFIG);
+    void runTestWithDriver(final TestRecord<Long, String> expectedFinalResult, final String storeName) throws InterruptedException {
```

Comment:
This is not introduced in this PR but in the existing `runTestWithDriver` we never use the passed in storeName it seems, and also the local `expectedFinalResult` are not used. Is that intentional?

Reply:
Okay, sounds fine.